Build failure: python38Packages.meson

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `python38Packages.meson`

### Build log

[python38Packages.meson.log](https://github.com/NixOS/nixpkgs/files/10829336/python38Packages.meson.log)

```
[1/2] Compiling C object ext/tachyon.so.p/tachyon_module.c.o
FAILED: ext/tachyon.so.p/tachyon_module.c.o 
gcc -Iext/tachyon.so.p -Iext '-I../test cases/python3/2 extmodule/ext' -I/nix/store/jb43gwp472xqwggcwjdvnc4lj26f8x6a-python3-3.8.16/include/python3.8 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fPIC -MD -MQ ext/tachyon.so.p/tachyon_module.c.o -MF ext/tachyon.so.p/tachyon_module.c.o.d -o ext/tachyon.so.p/tachyon_module.c.o -c '../test cases/python3/2 extmodule/ext/tachyon_module.c'
In file included from ../test cases/python3/2 extmodule/ext/tachyon_module.c:19:
/nix/store/jb43gwp472xqwggcwjdvnc4lj26f8x6a-python3-3.8.16/include/python3.8/Python.h:44:10: fatal error: crypt.h: No such file or directory
   44 | #include <crypt.h>
      |          ^~~~~~~~~
compilation terminated.
```

### Additional context

Happens on master (commit ee18200c318e7d62f7bdad3a9593b129d251e0db) and unstable (988cc958c57ce4350ec248d2d53087777f9e1949).

### Notify maintainers

@jtojnar @brandonedens @AndersonTorres 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.89-xanmod1, NixOS, 22.11 (Raccoon), 22.11.2523.1b82144edfc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(bt1cn): `"unstable"`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/python-modules/unrpa/default.nix
+++ b/pkgs/development/python-modules/unrpa/default.nix
@@ -6,14 +6,21 @@ buildPythonPackage rec {
 
   disabled = isPy27;
 
+  format = "setuptools";
+
   src = fetchPypi {
     inherit pname version;
     sha256 = "0yl4qdwp3in170ks98qnldqz3r2iyzil5g1775ccg98qkh95s724";
   };
 
-  propagatedBuildInputs = [ uncompyle6 ];
+  passthru.optional-dependencies = {
+    ZiX = [ uncompyle6 ];
+  };
 
   pythonImportsCheck = [ "unrpa" ];
+
+  # upstream has no unit tests
+  doCheck = false;
 
   meta = with lib; {
     homepage = "https://github.com/Lattyware/unrpa";

--- a/pkgs/development/python-modules/unrpa/default.nix
+++ b/pkgs/development/python-modules/unrpa/default.nix
@@ -1,12 +1,16 @@
-{ lib, buildPythonPackage, fetchPypi, uncompyle6, isPy27 }:
+{ lib
+, buildPythonPackage
+, fetchPypi
+, uncompyle6
+, pythonOlder
+}:
 
 buildPythonPackage rec {
   pname = "unrpa";
   version = "2.3.0";
-
-  disabled = isPy27;
-
   format = "setuptools";
+
+  disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
@@ -24,6 +28,7 @@ buildPythonPackage rec {
 
   meta = with lib; {
     homepage = "https://github.com/Lattyware/unrpa";
+    changelog = "https://github.com/Lattyware/unrpa/releases/tag/${version}";
     description = "A program to extract files from the RPA archive format";
     license = licenses.gpl3;
     maintainers = with maintainers; [ leo60228 ];

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -13030,7 +13030,7 @@ with pkgs;
 
   unrtf = callPackage ../tools/text/unrtf { };
 
-  unrpa = with python38Packages; toPythonApplication unrpa;
+  unrpa = with python3Packages; toPythonApplication unrpa;
 
   untex = callPackage ../tools/text/untex { };
 

